src/ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,615 @@
+            "IrfTool.full_enclosure",
+            "Compute a full-enclosure IRF.",
+            "Compute a point-like IRF.",
+        ),
+    }
+
+    classes = (
+        [
+            EventsLoader,
+        ]
+        + classes_with_traits(BackgroundRateMakerBase)
+        + classes_with_traits(EffectiveAreaMakerBase)
+        + classes_with_traits(EnergyMigrationMakerBase)
+        + classes_with_traits(PsfMakerBase)
+        + classes_with_traits(AngularResolutionMakerBase)
+        + classes_with_traits(EnergyBiasResolutionMakerBase)
+        + classes_with_traits(SensitivityMakerBase)
+    )
+
+    def setup(self):
+        self.opt_result = OptimizationResultStore().read(self.cuts_file)
+
+        if not self.full_enclosure and self.opt_result.theta_cuts is None:
+            raise ToolConfigurationError(
+                "Computing a point-like IRF requires an (optimized) theta cut."
+            )
+
+        check_e_bins = partial(
+            check_bins_in_range,
+            valid_range=self.opt_result.valid_energy,
+            raise_error=self.range_check_error,
+        )
+        self.particles = [
+            EventsLoader(
+                parent=self,
+                kind="gammas",
+                file=self.gamma_file,
+                target_spectrum=self.gamma_target_spectrum,
+            ),
+        ]
+        if self.do_background:
+            if self.proton_file:
+                self.particles.append(
+                    EventsLoader(
+                        parent=self,
+                        kind="protons",
+                        file=self.proton_file,
+                        target_spectrum=self.proton_target_spectrum,
+                    )
+                )
+            if self.electron_file:
+                self.particles.append(
+                    EventsLoader(
+                        parent=self,
+                        kind="electrons",
+                        file=self.electron_file,
+                        target_spectrum=self.electron_target_spectrum,
+                    )
+                )
+            if len(self.particles) == 1:
+                raise RuntimeError(
+                    "At least one electron or proton file required when specifying `do_background`."
+                )
+
+            self.bkg = BackgroundRateMakerBase.from_name(
+                self.bkg_parameterization, parent=self
+            )
+            check_e_bins(
+                bins=self.bkg.reco_energy_bins, source="background reco energy"
+            )
+
+        self.edisp = EnergyMigrationMakerBase.from_name(
+            self.edisp_parameterization, parent=self
+        )
+        self.aeff = EffectiveAreaMakerBase.from_name(
+            self.aeff_parameterization, parent=self
+        )
+
+        if self.full_enclosure:
+            self.psf = PsfMakerBase.from_name(self.psf_parameterization, parent=self)
+
+        if self.do_benchmarks:
+            self.b_output = self.output_path.with_name(
```

Comment:
please don't automatically create output files with names not under user control.

A good solution would be to replace the boolean `do_bechmarks` with an optional `benchmarks_output` Path and if it is given, compute the benchmarks and put them there.